api/test/integration/tavern_utils.py

```diff
@@ -328,3 +334,110 @@ def test_validate_search(response, search_param):
 
 def test_validate_key_not_in_response(response, key):
     assert all(key not in item for item in response.json()["data"]["affected_items"])
+
+
+def check_agentd_started(response, agents_list):
+    """Wait until all the agents had their agentd process started correctly. This will avoid race conditions caused by
+    agents reconnections before restarting.
+
+    Parameters
+    ----------
+    response : Request response
+    agents_list : list
+        List of expected agents to be restarted.
+    """
+
+    def get_timestamp(log):
+        """Get timestamp from log.
+
+        Parameters
+        ----------
+        log : str
+            String representing the log to get the timestamp from.
+
+        Returns
+        -------
+        datetime
+            Datetime object representing the timestamp got.
+        """
+        timestamp = re.search(r'^\d\d\d\d/\d\d/\d\d\s\d\d:\d\d:\d\d', log).group(0)
+        return datetime.strptime(timestamp, "%Y/%m/%d %H:%M:%S")
+
+    # Save the time when the restart command was sent
+    restart_request_time = datetime.utcnow().replace(microsecond=0) - response.elapsed
+
+    for agent_id in agents_list:
+        tries = 0
+        while tries < 80:
+            try:
+                # Save agentd logs in a list
+                command = f"docker exec env_wazuh-agent{int(agent_id)}_1 grep agentd /var/ossec/logs/ossec.log"
+                output = subprocess.check_output(command.split()).decode().strip().split('\n')
+            except subprocess.CalledProcessError:
+                assert False, f"Error while trying to get logs from agent {agent_id}"
+
+            # Ignore agentd logs before restart_request_time
+            logs_after_restart = [agentd_log for agentd_log in output if
+                                  get_timestamp(agentd_log).timestamp() >= restart_request_time.timestamp()]
+
+            # Check the log indicating agentd started is in the agent's ossec.log (after the restart request)
+            if any(re.search(pattern='agentd.*Started', string=agentd_log) for agentd_log in
+                   logs_after_restart):
+                break
+
+            tries += 1
+            time.sleep(1)
+        else:
+            assert False, "The wazuh-agentd daemon was not started after requesting the restart"
+
+
+def check_agent_active_status(agents_list):
+    """Wait until all the agents have active status in the global.db. This will avoid race conditions caused by
+    non-active agents in following test cases.
+
+    Parameters
+    ----------
+    agents_list : list
+        List of expected agents to be restarted.
+    """
+    id_active_agents = []
+    tries = 0
+    while tries < 25:
+        try:
+            # Get active agents
+            python_code = "from wazuh import agent; print(agent.get_agents(select=['status'], " \
+                          "filters={'status': 'active'})._affected_items)"
```

Comment:
I believe this should be saved in a python script, taking advantage of the docker volumes (all scripts will be shared in the same directory), to preserve maintainability and readability.

Reply:
Script added to `api/test/integration/env/tools`